### PR TITLE
acknowledge admin and course team on program page

### LINKED ISSUE #4430: Acknowledge admin and course team on program page
As a potential micromasters learner, I'd like to understand more about who teachers the courses and who oversees the program. 

As a member of the course team, I'd like to have my contribution to the courses acknowledged appropriately. 

As a member of the course team, I need to be able to update the names, titles, images and bios of all members at any time. 

#### Designs and Mockups

https://invis.io/Y8UKRZZSC6D#/391397475_MicroMasters_Team_Page_Copy (white background)

Sample copy at https://drive.google.com/file/d/1hYhbeNmzSVS-Hd3zBNzhcsRd2uefIlcJ/view

#### Acceptance Criteria:

- [x] Create a new tab specifically for the Course team, which probably requires a new wagtail page template
  - [x] The course team template should always be second
- [x] the course team page has four sections: Instructors, Administrative Team, Course Contributors and Current TAs. These section labels should have defaults but be able to be overridden. 
- [x] for each photo, there is a hover state that displays the person's bio
- [x] nice to have: ability to upload images from the page editor, instead of uploading them in the image section
- ~~add a link to this "course team" page below the existing faculty carousel (conditional on whether the page exists)~~

#### Questions

- **Q:** The current Instructor carousel is currently included in _every_ tab in the Program page. Should we reconsider this?
- **A:** I discussed with stakeholders, and we decided to keep it as is. 
- **Q:** What to do with bios?
- **A:** Bios will display on hover

### PR DESCRIPTION
**Pull request by tasawernawaz**
#### Pre-Flight checklist

- [x] Screenshots and design review for any changes that affect layout or styling
  - [ ] Desktop screenshots
  - [ ] Mobile width screenshots
  - [ ] Tag @ferdi or @pdpinch for review
- [x] Migrations
  - [ ] Migration is backwards-compatible with current production code
- [x] Testing
  - [ ] Code is tested
  - [ ] Changes have been manually tested
- [ ] Settings
  - [ ] New settings are documented and present in `app.json`
  - [ ] New settings have reasonable development defaults, if applicable

#### What are the relevant tickets?
Fixes #4430 

#### What's this PR do?
Add course team tab page for program page.

#### How should this be manually tested?

- Goto cms
- Add child page "Course team"
- Fill the different sections on the page.
- Goto to program page
- You should see a new tab page for the course team.

#### Screenshots (if appropriate)
**CMS**
68310402-c4e92500-00d1-11ea-9e05-d8ff7e9e105e

**Rendering**
screencapture-localhost-8079-my-program-page-course-team-2019-11-08-19_40_30

### CODE REVIEW

**Comment by tasawernawaz** [comment]
@pdpinch Could you please take a quick look at the design, you might want to change the font color of the instructor's name on the image.

**Comment by tasawernawaz** [comment]
FYI @abdulkdawson ^

**Comment by abdulkdawson** [comment]
@tasawernawaz what are the image dimensions?

**Comment by abdulkdawson** [comment]
@tasawernawaz Also can we add a black gradient to the bottom of the image so that the name is clearer. I'd rather do that then change the font color. It can be just as problematic if we do that.

**Comment by pdpinch** [comment]
The Instructor images should be larger than the administrator images. @tasawernawaz Have you been able to access [inspect mode](https://projects.invisionapp.com/d/main/#/console/18769740/391397475/inspect)?

**Comment by tasawernawaz** [comment]
@pdpinch @abdulkdawson Sorry I missed the difference in profile image of instructors and administrators. I have updated the design with the given dimensions of the image and added gradient too. However, the instructor section given dimension is 385x385 which can only be seen when you have a high-resolution screen. New screenshot attached below.

screencapture-localhost-8079-my-program-page-course-team-2019-11-11-18_20_42

**Comment by tasawernawaz** [comment]
@pdpinch I have following question about adding `a link to this "course team" page below the existing faculty carousel (conditional on whether the page exists)`

Q#1: Will link to the course team be on every tab as we have faculty carousel on every tab.
Q#2: What should the be design to add a link to the course team.

File: cms/blocks.py [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+from wagtail.images.blocks import ImageChooserBlock
+
+
+class CourseTeamBlock(blocks.StructBlock):
+    """
+    Block class that defines a course team member
+    """
+
+    name = blocks.CharBlock(max_length=100, help_text="Name of the course team member.")
+    bio = blocks.CharBlock(max_length=255, help_text="Short bio of course team member.")
```
Comment (ahmed-arbisoft) [inline cms/blocks.py]: Are we okay with the 255 character limit on this?

**Comment by tasawernawaz** [comment]
@pdpinch I know that @annagav is on leave, who else can review this? @ahmed-arbisoft has reviewed but he cannot do a functional test as he doesn't have MM locally.

**Comment by pdpinch** [comment]
We decided not to do [the link](https://github.com/mitodl/micromasters/pull/4454#issuecomment-552831602) for now. If we still need to do it, I'll open another issue. 

I updated the issue description.

**Comment by tasawernawaz** [comment]
> We decided not to do [the link](https://github.com/mitodl/micromasters/pull/4454#issuecomment-552831602) for now. If we still need to do it, I'll open another issue.
> 
> I updated the issue description.

thanks, what else do think required on this before we can merge this one.

File: cms/blocks.py (hunk 1 above)
Comment (tasawernawaz) [reply cms/blocks.py]: @pdpinch could you please answer this question.

**Comment by pdpinch** [comment]
Can you set up the review app so I can interact with the page? I'd like to see how the bios work on hover.

**Comment by tasawernawaz** [comment]
> Can you set up the review app so I can interact with the page? I'd like to see how the bios work on hover.

sure thing.

**Comment by tasawernawaz** [comment]
@pdpinch I have setup a review app for testing.
You can review at https://micromasters-ci-pr-4454.herokuapp.com/test-course-team-program/
You can edit at https://micromasters-ci-pr-4454.herokuapp.com/cms/pages/5/edit/

User: `staff`
password: `edx`

**Comment by annagav** [comment]
@tasawernawaz, I noticed that when you hover over the picture the name of faculty changes font. Is that intentional? 
Also, something feels abrupt about the hover. Is it possible to make the fade out of the picture more smooth. If this feature is urgent, then it's okay as it is.

**Comment by pdpinch** [comment]
@annagav I get an animation of the bio when I hover -- it scrolls up. What browser are you using?

File: cms/blocks.py (hunk 1 above)
Comment (pdpinch) [reply cms/blocks.py]: Yes, definitely. The size of the input box in the CMS is too small.

**Comment by annagav** [comment]
@pdpinch yes, I can see the hover, but I thought something looked out of sync , or happens too fast. Like the fade out of the image good have a smoother transition, the size of the font changes abruptly.

File: cms/blocks.py (hunk 1 above)
Comment (pdpinch) [reply cms/blocks.py]: Also, 255 characters is not enough for the existing sample copy at https://drive.google.com/file/d/1hYhbeNmzSVS-Hd3zBNzhcsRd2uefIlcJ/view 

The longest bio there looks to me to be about 700 characters. I'm not sure if this breaks the layout.

**Comment by pdpinch** [comment]
Please make the layout responsive, and include a screenshot at mobile width:

This isn't a good look: 

image

**Comment by pdpinch** [comment]
Also, I'm not sure why we have so much padding between the course administrator images. It would be nice to allow them to go to four across at desktop screen size.

**Comment by pdpinch** [comment]
The course team tab is displayed on the "about the progra" page, but it disappears on the "course team" page. It should change to a selected state, similar to what the FAQ does on the current site. 

image

**Comment by pdpinch** [comment]
@tasawernawaz for Anna's [animation question](https://github.com/mitodl/micromasters/pull/4454#issuecomment-553559032), can we slow it down?

**Comment by tasawernawaz** [comment]
@pdpinch @annagav I can add animation, but difference in the font and padding between administrators are intentional as they are given in the design https://invis.io/Y8UKRZZSC6D#/391397475_MicroMasters_Team_Page_Copy.

Honestly, It wasn't an easy design, I have to took help from a designer here to match the exact padding on different sections of the page.

**Comment by tasawernawaz** [comment]
@annagav , @pdpinch I have addressed the following comments

- added opacity to handle hover state better.
- updated administrators section to have four items
- Fix active tab issue
- Fix for mobile and tablet

**Desktop**
screencapture-mm-odl-local-8079-test-tasawer-program-course-team-2019-11-14-15_45_58

**Tablet**
screencapture-mm-odl-local-8079-test-tasawer-program-course-team-2019-11-14-15_49_02


**Mobile**
screencapture-mm-odl-local-8079-test-tasawer-program-course-team-2019-11-14-15_59_10

**Pull request merged**